Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaStreamAVFObjC.mm

```diff
@@ -750,19 +757,26 @@ void getSupportedTypes(HashSet<String>& types) const final
 
 void MediaPlayerPrivateMediaStreamAVFObjC::characteristicsChanged()
 {
-    bool sizeChanged = false;
+    SizeChanged sizeChanged = SizeChanged::No;
 
     FloatSize intrinsicSize = m_mediaStreamPrivate->intrinsicSize();
-    if (intrinsicSize.height() != m_intrinsicSize.height() || intrinsicSize.width() != m_intrinsicSize.width()) {
-        m_intrinsicSize = intrinsicSize;
-        sizeChanged = true;
-        if (m_playbackState == PlaybackState::None)
-            m_playbackState = PlaybackState::Paused;
+    if (intrinsicSize.isEmpty() || m_intrinsicSize.isEmpty()) {
+        if (intrinsicSize.height() != m_intrinsicSize.height() || intrinsicSize.width() != m_intrinsicSize.width()) {
+            m_intrinsicSize = intrinsicSize;
+            sizeChanged = SizeChanged::Yes;
+            if (m_playbackState == PlaybackState::None)
+                m_playbackState = PlaybackState::Paused;
+        }
     }
 
     updateTracks();
     updateDisplayMode();
 
+    scheduleTaskForCharacteristicsChanged(sizeChanged);
```

Comment:
do we need to pass sizeChanged?

why would we call scheduleTaskForCharacteristicsChanged if the size didn't change?
if the size was empty, we would fire with SizeChanged.

And if the size didn't change, there's no need to updateLayers right?